### PR TITLE
Encode wide char to CP_ACP for CLR usage

### PR DESCRIPTION
**Pull request by schellap**
Fixes https://github.com/dotnet/cli/issues/2649

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2673)

### CODE REVIEW

File: src/corehost/cli/hostpolicy.cpp [hunk 1]
```diff
@@ -147,28 +148,31 @@ int run(const arguments_t& args)
...
+        for (int i = 0; i < argv.size(); i++)
         {
-            arg_str.append(args.app_argv[i]);
+            pal::string_t cur;
+            pal::clr_palstring(argv[i], &cur);
+            arg_str.append(cur);
             arg_str.append(_X(","));
         }
-        trace::info(_X("Launch host: %s app: %s, argc: %d args: %s"), args.own_path.c_str(),
+        trace::info(_X("Launch, host: %s app: %s, argc: %d args: %s"), args.own_path.c_str(),
```
Comment (eerhardt) [inline src/corehost/cli/hostpolicy.cpp]: Was the added comma intentional?

**Comment by eerhardt** [comment]
:shipit:

File: src/corehost/cli/hostpolicy.cpp (hunk 1 above)
Comment (schellap) [inline src/corehost/cli/hostpolicy.cpp]: No it was added while debugging to detect change. Intent was not to commit it. Thanks.

**Comment by schellap** [comment]
@gkhanna79, would appreciate a look. @eerhardt confirmed that this PR is good for what he was trying to do.

**Comment by gkhanna79** [comment]
LGTM

**Comment by schellap** [comment]
@eerhardt, just waiting for OSX.

**Comment by schellap** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by piotrpMSFT** [comment]
OS X queue is backed up...

**Pull request merged**